flink-runtime/src/main/java/org/apache/flink/runtime/resourcemanager/slotmanager/ResourceActions.java

```diff
@@ -53,4 +56,13 @@
 	 * @param cause of the allocation failure
 	 */
 	void notifyAllocationFailure(JobID jobId, AllocationID allocationId, Exception cause);
+
+	/**
+	 * Notifies that the task manager has been terminated.
+	 * @param jobId to be notified
+	 * @param resourceID identifying the terminated task manager
+	 * @param allocationIDs of the job held that belong to this task manager
+	 * @param cause of the task manager termination.
+	 */
+	void notifyTaskManagerTerminated(JobID jobId, ResourceID resourceID, Set<AllocationID> allocationIDs, Exception cause);
```

Comment:
I think the notification about a terminated `TaskManager` should not come from the `SlotManager` but from the `ResourceManager`. Thus, we should not need this method.